src/main/scala/fs2/aws/kinesis/kcl.scala

```diff
@@ -0,0 +1,90 @@
+package fs2
+package aws
+package kinesis
+
+import fs2.internal._
+import fs2.concurrent.Queue
+import cats.effect.{ConcurrentEffect, IO, Timer}
+import cats.implicits._
+
+import com.amazonaws.services.kinesis.model.Record
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.Worker
+import com.amazonaws.services.kinesis.clientlibrary.interfaces.v2.IRecordProcessorFactory
+
+package object kcl {
+  /** Intialize a worker and start streaming records from a Kinesis stream
+    * On stream finish (due to error or other), worker will be shutdown
+    *
+    *  @tparam F effect type of the fs2 stream
+    *  @param workerFactory function to create a Worker from a IRecordProcessorFactory
+    *  @param config configuration settings for the KCL worker
+    *  @return an infinite fs2 Stream that emits Kinesis Records
+    */
+  def readFromKinesisStream[F[_]](
+    workerFactory: => IRecordProcessorFactory => Worker,
```

Comment:
I'm not familiar with Kinesis enough but is there a way we could provide a default workerFactory ? or abstract completely the need for a factory in the first place ? Factory sounds very Javaish

Reply:
Good idea! I'll play around with something a bit more idiomatic